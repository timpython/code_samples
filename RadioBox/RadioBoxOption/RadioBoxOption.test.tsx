import React from 'react';
import { shallow } from 'enzyme';
import { RadioBoxOption } from '.';

describe('Option', () => {
  it('renders correctly', () => {
    const onChange = () => {};
    const content = <div>Some Content</div>;

    const wrapper = shallow(
      <RadioBoxOption label="text" name="name" value={0} onChange={onChange} checked={true} content={content} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('onChange is called properly', () => {
    const onChange = jest.fn();
    const eventObject = {
      name: 'name',
      value: 'value',
    };

    const wrapper = shallow(
      <RadioBoxOption label="text" name="name" value="value" onChange={onChange} checked={true} />
    );
    wrapper.find('input').simulate('change', eventObject);
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(eventObject);
  });
});
