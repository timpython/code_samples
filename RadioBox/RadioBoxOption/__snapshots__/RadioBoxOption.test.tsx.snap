// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option renders correctly 1`] = `
<div
  className="container"
>
  <label
    className="label"
  >
    <input
      checked={true}
      className="radioButton"
      name="name"
      onChange={[Function]}
      type="radio"
      value={0}
    />
    <span
      className="labelText"
    >
      text
    </span>
    <span
      className="checkMark"
    />
  </label>
  <div
    className="content"
  >
    <div>
      Some Content
    </div>
  </div>
</div>
`;
